--- b/build/config/compiler/BUILD.gn
+++ a/build/config/compiler/BUILD.gn
@@ -2197,6 +2197,9 @@
   # an error by default eventually; see
   # https://github.com/rust-lang/rust/issues/71668
   rustflags = [ "-Dunsafe_op_in_unsafe_fn" ]
+
+  # TODO(crbug.com/424424323): Clean up and enable.
+  rustflags += [ "-Amismatched_lifetime_syntaxes" ]
 }
 
 # prevent_unsafe_narrowing ----------------------------------------------------
